jolt-core/src/main/java/com/bazaarvoice/jolt/templatr/function/Function.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2013 Bazaarvoice, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bazaarvoice.jolt.templatr.function;
+
+import com.bazaarvoice.jolt.common.Optional;
+import com.bazaarvoice.jolt.exception.SpecException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public interface Function {
+
+    public Optional<Object> apply(Object... args);
+
+    public static final class Registry {
+
+        private static final Map<String, Function> INSTANCES = new HashMap<>();
+
+        static {
+            Function.Registry.register("toLower", new Strings.toLowerCase() );
+            Function.Registry.register("toUpper", new Strings.toUpperCase() );
+            Function.Registry.register("toProper", new Strings.toProperCase() );
+            Function.Registry.register("strConcat", new Strings.strConcat() );
+            Function.Registry.register("maxOf", new Math.MaxOf() );
+            Function.Registry.register("minOf", new Math.MinOf() );
+            Function.Registry.register("abs", new Math.Abs() );
+            Function.Registry.register("toInteger", new Math.toInteger() );
+            Function.Registry.register("toDouble", new Math.toDouble() );
+        }
+
+        public static Function register(String className, Function instance) {
```

Comment:
static registries are bad, means there is only one per JVM.   Go with the Dropwizard7 "Metrics Registry" were there is a default one that is static, but that you can also make one and use your own / or at least NameSpace it.